prover/crates/lib/circuit_prover_service/src/gpu_circuit_prover/gpu_circuit_prover_executor.rs

```diff
@@ -0,0 +1,73 @@
+use std::time::Instant;
+
+use anyhow::Context;
+use shivini::ProverContext;
+use zksync_prover_fri_types::FriProofWrapper;
+use zksync_prover_job_processor::Executor;
+use zksync_types::prover_dal::FriProverJobMetadata;
+
+use crate::{
+    metrics::CIRCUIT_PROVER_METRICS, types::circuit_prover_payload::GpuCircuitProverPayload,
+};
+
+/// GpuCircuitProver executor implementation.
+/// Generates circuit proof & verifies it.
+/// NOTE: It requires prover context, which is the way Shivini allocates VRAM.
+pub struct GpuCircuitProverExecutor {
```

Comment:
Nit(I don't see a reason to have it as a separate field):
```suggestion
pub struct GpuCircuitProverExecutor(ProverContext);
```